app/controllers/preauthorize_transactions_controller.rb

```diff
@@ -547,20 +561,30 @@ def ensure_can_receive_payment
   end
 
   def create_preauth_transaction(opts)
-
-    # PayPal doesn't like images with cache buster in the URL
-    logo_url = Maybe(opts[:community])
-                 .wide_logo
-                 .select { |wl| wl.present? }
-                 .url(:paypal, timestamp: false)
-                 .or_else(nil)
-
-    gateway_fields =
-      {
-        merchant_brand_logo_url: logo_url,
-        success_url: success_paypal_service_checkout_orders_url,
-        cancel_url: cancel_paypal_service_checkout_orders_url(listing_id: opts[:listing].id)
-      }
+    case opts[:payment_type].to_sym
+    when :paypal
+      # PayPal doesn't like images with cache buster in the URL
+      logo_url = Maybe(opts[:community])
+               .wide_logo
+               .select { |wl| wl.present? }
+               .url(:paypal, timestamp: false)
+               .or_else(nil)
+
+      gateway_fields =
+        {
+          merchant_brand_logo_url: logo_url,
+          success_url: success_paypal_service_checkout_orders_url,
+          cancel_url: cancel_paypal_service_checkout_orders_url(listing_id: opts[:listing].id)
+        }
+    when :stripe
+      gateway_fields =
+        {
+          stripe_email: @current_user.primary_email.address,
```

Comment:
1) this is the buyers email, a buyer will not have a stripe custom account
2) adding metadata information is a good idea, will add this to a separate task